samples/core/Miscellaneous/NewInEFCore7/DocumentsContext.cs

```diff
@@ -0,0 +1,656 @@
+    public decimal? CoverPrice { get; set; }
+    public Person Editor { get; set; } = null!;
+}
+
+public class Person
+{
+    public Person(string name)
+    {
+        Name = name;
+    }
+
+    public int Id { get; private set; }
+
+    [ConcurrencyCheck]
+    public string Name { get; set; }
+
+    public ContactDetails Contact { get; set; } = null!;
+
+    public List<Book> PublishedWorks { get; } = new();
+    public List<Magazine> Edited { get; } = new();
+}
+
+public abstract class DocumentsContext : DbContext
+{
+    public bool LoggingEnabled { get; set; }
+    public abstract MappingStrategy MappingStrategy { get; }
+    public virtual bool UseStoredProcedures => true;
+
+    public DbSet<Document> Documents => Set<Document>();
+    public DbSet<Book> Books => Set<Book>();
+    public DbSet<Magazine> Magazines => Set<Magazine>();
+    public DbSet<Person> People => Set<Person>();
+
+    protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
+        => optionsBuilder.UseSqlServer(@$"Server=(localdb)\mssqllocaldb;Database={GetType().Name}")
+            .EnableSensitiveDataLogging()
+            .LogTo(
+                s =>
+                {
+                    if (LoggingEnabled)
+                    {
+                        Console.WriteLine(s);
+                    }
+                }, LogLevel.Information);
+
+    protected override void OnModelCreating(ModelBuilder modelBuilder)
+    {
+        modelBuilder.Entity<Document>(
+            entityTypeBuilder =>
+            {
+                entityTypeBuilder.Property(document => document.FirstRecordedOn).HasDefaultValueSql("getutcdate()");
+                entityTypeBuilder.Property(document => document.RetrievedOn).HasComputedColumnSql("getutcdate()");
+            });
+
+        modelBuilder.Entity<Person>(
+            entityTypeBuilder =>
+            {
+                if (UseStoredProcedures)
+                {
+                    entityTypeBuilder.InsertUsingStoredProcedure(
+                        "Person_Insert", storedProcedureBuilder =>
+                        {
+                            storedProcedureBuilder.HasParameter(a => a.Name);
+                            storedProcedureBuilder.HasResultColumn(a => a.Id);
+                        });
+
+                    entityTypeBuilder.UpdateUsingStoredProcedure(
+                        "Person_Update",
+                        storedProcedureBuilder =>
+                        {
+                            storedProcedureBuilder.HasParameter(person => person.Id);
+                            storedProcedureBuilder.HasOriginalValueParameter(
+                                person => person.Name, parameterBuilder => parameterBuilder.HasName("Name_Original"));
+                            storedProcedureBuilder.HasParameter(person => person.Name);
+                            storedProcedureBuilder.HasRowsAffectedResultColumn();
+                        });
```

Comment:
BTW any reason not to just chain calls in a single expression here and everywhere else?

```suggestion
                    entityTypeBuilder.UpdateUsingStoredProcedure(
                        "Person_Update",
                        storedProcedureBuilder => storedProcedureBuilder
                            .HasParameter(person => person.Id)
                            .HasOriginalValueParameter(person => person.Name, parameterBuilder => parameterBuilder.HasName("Name_Original"))
                            .HasParameter(person => person.Name)
                            .HasRowsAffectedResultColumn());
```

Reply:
No reason, other than I added them one at a time and didn't think about it. Will do.